<html>
<head>
<script src="../../http/tests/inspector-protocol/resources/protocol-test.js"></script>
<script src="resources/breakpoint.js"></script>
<script>
function test()
{
    InspectorTest.eventHandler["Debugger.scriptParsed"] = function(messageObject)
    {
        if (/resources\/breakpoint\.js$/.test(messageObject.params.url)) {
            InspectorTest.log("Found breakpoint.js");
            var breakpoint = {
                location: {
                    scriptId: messageObject.params.scriptId,
                    lineNumber: 3,
                    columnNumber: 0
                }
            };
            InspectorTest.sendCommand("Debugger.setBreakpoint", breakpoint, InspectorTest.checkForError);
            InspectorTest.sendCommand("Debugger.setPauseOnExceptions", {state: "all"}, InspectorTest.checkForError);
            InspectorTest.sendCommand("Runtime.evaluate", {expression: "breakpointBasic()"}, function() {
                completeTest();
            });
        }
    }

    InspectorTest.eventHandler["Debugger.paused"] = onBreakpointHit;

    function onBreakpointHit(messageObject)
    {
        InspectorTest.log("PASS: Paused in debugger: reason = \"" + messageObject.params.reason + "\"");

        // Make sure we do not pause again in the debugger.
        InspectorTest.eventHandler["Debugger.paused"] = onPausedOnException;

        InspectorTest.sendCommand("Runtime.evaluate", {expression: "this.should.trigger.an.exception"}, function(messageObject) {
            InspectorTest.checkForError(messageObject);
            InspectorTest.log("PASS: The debugger should not stop on the exception thrown by a console command while paused on a breakpoint.");
            InspectorTest.sendCommand("Debugger.resume");
        });
    }

    function onPausedOnException(messageObject)
    {
        InspectorTest.log("FAIL: Should not break on exceptions while paused on a breakpoint.");
        completeTest();
    }

    function completeTest()
    {
        // InspectorTest.sendCommand("Debugger.resume", {});
        // Reset the pauseOnException state before ending the test.
        InspectorTest.sendCommand("Debugger.setPauseOnExceptions", {state: "none"});
        InspectorTest.sendCommand("Debugger.disable");
        InspectorTest.completeTest();
    }

    InspectorTest.sendCommand("Debugger.enable", {});
}
</script>
</head>
<body onload="runTest()">
<p>While being paused on a breakpoint the debugger should ignore the exceptions from console commands.</p>
</body>
</html>
